agent/api/task/task.go

```diff
@@ -525,6 +526,20 @@ func (task *Task) initServiceConnectContainerDependencies() {
 		container.AddContainerDependency(scContainer.Name, ContainerOrderingHealthyCondition)
 		scContainer.BuildContainerDependency(container.Name, apicontainerstatus.ContainerStopped, apicontainerstatus.ContainerStopped)
 	}
+
+	// For bridge mode, we are going to have the SC container configure netns for itself as well as for all other task
+	// containers. The configuration is done when SC container transitions from RUNNING -> RESOURCES_PROVISIONED.
+	// Therefore, each application container will also wait for scContainer.RESOURCE_PROVISIONED stage to complete
+	// before they can be created (i.e. application container will pause at the previous stage which is PULLED.
+	if task.IsNetworkModeBridge() {
```

Comment:
Sure. The reason why I had a separate block if block was because we also need to override steady status for SC container so I thought to group them together for readability. I don't have strong preference though so yeah I'll get rid of two loops.